pkg/resources/cloudcontroller/digitalocean.go

```diff
@@ -0,0 +1,129 @@
+			}
+
+			dep.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+				Labels: podLabels,
+			}
+
+			dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err =
+				resources.UserClusterDNSPolicyAndConfig(data)
+			if err != nil {
+				return nil, err
+			}
+
+			dep.Spec.Template.Spec.AutomountServiceAccountToken = pointer.Bool(false)
+
+			dep.Spec.Template.Spec.Volumes = append(getVolumes(data.IsKonnectivityEnabled()),
+				corev1.Volume{
+					Name: resources.CloudConfigConfigMapName,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: resources.CloudConfigConfigMapName,
+							},
+						},
+					},
+				})
+
+			dep.Spec.Template.Spec.Containers = []corev1.Container{
+				{
+					Name:    ccmContainerName,
+					Image:   data.ImageRegistry(resources.RegistryMCR) + "/oss/kubernetes/digital-ocean-cloud-controller-manager:v" + DigitalOceanVersion,
+					Command: []string{"cloud-controller-manager"},
+					Args:    getDigitalOceanFlags(data),
+					VolumeMounts: append(getVolumeMounts(),
+						corev1.VolumeMount{
+							Name:      resources.CloudConfigConfigMapName,
+							MountPath: "/etc/kubernetes/cloud",
+							ReadOnly:  true,
+						},
+					),
+					LivenessProbe: &corev1.Probe{
+						ProbeHandler: corev1.ProbeHandler{
+							HTTPGet: &corev1.HTTPGetAction{
+								Scheme: corev1.URISchemeHTTP,
+								Path:   "/healthz",
+								Port:   intstr.FromInt(10267),
+							},
+						},
+						SuccessThreshold:    1,
+						FailureThreshold:    3,
+						InitialDelaySeconds: 20,
+						PeriodSeconds:       10,
+						TimeoutSeconds:      5,
+					},
+				},
+			}
+
+			defResourceRequirements := map[string]*corev1.ResourceRequirements{
+				ccmContainerName: azureResourceRequirements.DeepCopy(),
+			}
+
+			if !data.IsKonnectivityEnabled() {
+				openvpnSidecar, err := vpnsidecar.OpenVPNSidecarContainer(data, openvpnClientContainerName)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get openvpn sidecar: %w", err)
+				}
+				dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers, *openvpnSidecar)
+				defResourceRequirements[openvpnSidecar.Name] = openvpnSidecar.Resources.DeepCopy()
+			}
+
+			err = resources.SetResourceRequirements(dep.Spec.Template.Spec.Containers, defResourceRequirements, nil, dep.Annotations)
+			if err != nil {
+				return nil, fmt.Errorf("failed to set resource requirements: %w", err)
+			}
+			return dep, nil
+		}
+	}
+}
+
+func getDigitalOceanFlags(data *resources.TemplateData) []string {
+	flags := []string{
+		"--kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig",
+		"--leader-elect=true",
+		"--v=5",
+		"--cloud-config=/etc/kubernetes/cloud/config",
+		"--cloud-provider=digitalocean",
+	}
+
+	if data.Cluster().Spec.Features[kubermaticv1.ClusterFeatureCCMClusterName] {
+		flags = append(flags, "--cluster-name", data.Cluster().Name)
+	}
```

Comment:
I don't think we need to feature gate this flag since this is a new CCM. This feature gate is used as mitigation for providers where we added CCM support, but didn't put this flag from the beginning. Those clusters had to be manually migrated, and that's why we added this flag. That's not needed for DO.